02-Context-or-Narrative/State-of-PureScript-Documentation-2019.md

```diff
@@ -170,26 +170,158 @@ Here's what has been added to the above since then:
 
 ## Why is PureScript's Documentation Lacking and How Do We Improve It?
 
-- Philosophy of the language:
-    - avoid (success at all costs)
-    - wrong intepretation:
-    - right interpretation
-- Current priority of core contributors: adding powerful features via breaking changes
-    - wrong interpretation: they don't care
-    - right interpretation: want to improve language
-    - Why a `v1.0` hasn't been released
-- Core Contributors are spread thin, so they cannot respond quickly to things
-    - The '75 libraries' problem
-    - Entrusting library maintenance to others and supporting them (e.g. the `event-stream` incident)
-    - Just like reviewing code, reviewing documentation takes a lot of thought
-    - Hard to keep documentation momentum
-    - Which libraries are 'core libraries'?
-- No clearly-defined mutually-held community-wide goal
-    - A desire to help exists, but no one is coordinating resources strategically
-- What exactly are FP's "Best Practices?"
-    - They are not well-defined, are assumed, or are habits
-    - Knowledge silos exist, so it's hard to extract that info
-- Slack-based discussions do not persist
+These are the reasons. Each wil be covered more in-depth following this:
+
+- PureScript is not _currently_ trying to be the next "mainstream language".
+- Core contributors are spread thin: limited time with too much responsibility
+- There is no roadmap that coordinates efforts
+- It's difficult to pin down and define what are Functional Programming's "best practices"
+- Questions answered on Slack do not persist, so they get re-asked and re-answered
```

Comment:
Perhaps this idea is better phrased as a more general problem, rather than a specific one. Suggested alternatives:
- "The current support system is pull rather than push"
- "The current support system doesn't build towards structured, persistent documentation"
- "No process for contributing towards a learner-friendly education system"

Reply:
Hmm... I was trying to convey two points:
- questions do not persist: one cannot refer to them or look at them again later
- developers' time is wasted re-answering the same questions

> "The current support system is pull rather than push"

That doesn't mean a whole lot to mean when I read it. 'support system' is overloaded with other possible things that we don't intend here.

> "The current support system doesn't build towards structured, persistent documentation"

I think this gets closer to the heart of the issue. Docs aren't persistent; they aren't structured; and the 'process' we use doesn't move towards that goal. 

The issue arises largely in the Slack channel, but I agree that the issue can appear in other PureScript mediums of communication.

> "No process for contributing towards a learner-friendly education system"

What would be the points you would use to summarize this header if we used it? I think your suggestions have made me realize that I might be emphasizing the non-persistence Slack issue too much (a possible symptom) rather than the heart of the issue.

Reply:
I might write a section roughly like the following.

-----

## The current support system doesn't build towards structured, persistent documentation

Currently, the most active PureScript support channel is the #purescript channel on the functionalprogramming.slack.com chatroom, though it's also worth mentioning that PureScript's Discourse instance, discourse.purescript.org, is relatively active. It's not clear why one would choose one over the other, but a chatroom arguably is a more social, interactive experience than an internet forum like Discourse.

While a considerable portion of the PureScript support network seem to be motivated by the social aspect of a chatroom, it's possible that questions answered here means the question is not answered in a more accessible, educational format which supports users of the language motivated for reasons other than social connection, such as socially shy or productivity-motivated users. As a result, this organizational knowledge accumulated over time is locked in the heads of existing users and is harder for learners or future maintainers to access for their own education.

This problem can be alleviated by promoting a culture of monitoring the chatroom for conversations containing information which could be adapted for long-term storage in structured documentation targeting a certain audience.

Reply:
I only offered up one way of morphing the current support system into a structured documentation set, but I think we should discover more. Perhaps we can recommend newcomers, on the PS website and the chatroom header, to first read a short document explaining the documentation situation and that they are in a unique position to help. That document could, perhaps, recommend interacting with the learner-targeting documentation project as well.

Reply:
> Perhaps we can recommend newcomers, on the PS website and the chatroom header, to first read a short document explaining the documentation situation and that they are in a unique position to help. That document could, perhaps, recommend interacting with the learner-targeting documentation project as well.

That's kind of what I hope this document to be, something everyone can refer new people to to help them know what's going on immediately rather than 3 months after struggling through things.

Reply:
Regarding your attempt at writing the section:
- highlighting the point of a chatroom has a greater appeal than a place like the Discourse forum
- highlights the downside of a chatroom-format: people use it for non-Q&A stuff, like goofing around
- the 'solution' to the problem you propose is sort of what you and I are already doing: copying and pasting helpful things into Discourse or my own learning repo. My point is that it brings awareness to this activity. Perhaps it can encourage others to do likewise when you or I miss something? Still, one thing that isn't mentioned here is privacy. What gets posted on Slack is somewhat good because it will be deleted later, so some people are more open to share things there. If we copy and paste that elsewhere, then it loses some of that privacy.

Reply:
Right, I think that solution is the only one that would work.

> > "The current support system doesn't build towards structured, persistent documentation"

> I think this gets closer to the heart of the issue. Docs aren't persistent; they aren't structured; and the 'process' we use doesn't move towards that goal.

After looking at this again with fresh eyes, I think this is the best way to summarize it. That is because we are saying the primary issue is that the process for moving stuff from the primary support channel (a chatroom) into the structured persistent documentation isn't carried out as often as it should be.

Reply:
I've updated this PR to use that rendering for the header

Reply:
Looks good. 👌